pkg/status/controller.go

```diff
@@ -204,6 +204,14 @@ func (c *Controller) handleObject(obj any) {
 	c.enqueueObject(obj)
 }
 
+// handleObjectDeletion is like handleObject except that handleObjectDeletion calls
+// enqueueObjectDeletion instead of enqueueObject
+func (c *Controller) handleObjectDeletion(obj any) {
+	rObj := obj.(runtime.Object)
```

Comment:
This could fail; the object could be a https://github.com/kubernetes/client-go/blob/v0.28.4/tools/cache/delta_fifo.go#L800

BTW: If you stop enqueuing whole objects then you will be able to use https://github.com/kubernetes/client-go/blob/v0.30.0/tools/cache/controller.go#L342 when using Kube release 1.30 or later.